docs/source/onedal/algorithms/ensembles/decision-forest.rst

```diff
@@ -0,0 +1,218 @@
+each value from initially provided data with the value of the corresponding bin. Bins are continuous intervals between selected splits.
+
+.. _df_i_math:
+.. _df_i_math_dense:
+.. _df_i_math_hist:
+
+Inference methods: *Dense* and *Hist*
+-------------------------------------
+"Dense" and "Hist" inference methods performs prediction by the same way:
+
+#. For classification, :math:`y_i \in \{0, \ldots, \mathrm{C-1}\}`, where :math:`C` is the number of classes,
+   the tree ensemble model predicts the output by selecting the response :math:`y`,
+   which is voted for by the majority of the trees in the forest.
+
+#. For regression, the tree ensemble model uses the mean of :math:`M` functions' results to predict the
+   output, i.e. :math:`\hat{y}=\frac{1}{M} \sum_{k=1}^M{f_k(x_i)}, \; f_k \in F,` where :math:`F=\{f : \mathbb{R}^p \rightarrow W, \; W \subset \mathbb{R}, \; |W|=T\}` is a set of 
+   regression trees, :math:`W` is a set of tree leaves' scores and :math:`T` is the number of leaves in the tree.
+   In other words, each tree maps an observation to the corresponding leaf's score.
+
+---------------------
+Programming Interface
+---------------------
+All types and functions in this section shall be declared in the
+``oneapi::dal::decision_forest`` namespace and be available via inclusion of the
+``oneapi/dal/algo/decision_forest.hpp`` header file.
+                   
+Enum classes
+------------
+.. onedal_enumclass:: oneapi::dal::decision_forest::v1::error_metric_mode 
+.. onedal_enumclass:: oneapi::dal::decision_forest::v1::variable_importance_mode 
+.. onedal_enumclass:: oneapi::dal::decision_forest::v1::infer_mode
+.. onedal_enumclass:: oneapi::dal::decision_forest::v1::voting_mode
+                   
+Descriptor
+----------
+.. onedal_class:: oneapi::dal::decision_forest::detail::v1::descriptor_base
+.. onedal_class:: oneapi::dal::decision_forest::v1::descriptor
+
+Method tags
+~~~~~~~~~~~
+.. onedal_tags_namespace:: oneapi::dal::decision_forest::method::v1
+
+Task tags
+~~~~~~~~~
+.. onedal_tags_namespace:: oneapi::dal::decision_forest::task::v1
+
+Model
+-----
+.. onedal_class:: oneapi::dal::decision_forest::v1::model
+
+.. _df_t_api:
+
+Training :expr:`train(...)`
+--------------------------------
+.. _df_t_api_input:
+
+Input
+~~~~~
+.. onedal_class:: oneapi::dal::decision_forest::v1::train_input
+
+
+.. _df_t_api_result:
+
+Result
+~~~~~~
+.. onedal_class:: oneapi::dal::decision_forest::v1::train_result
+
+Operation
+~~~~~~~~~
+.. onedal_func:: oneapi::dal::v1::train
+
+.. _df_i_api:
+
+Inference :expr:`infer(...)`
+----------------------------
+.. _df_i_api_input:
+
+Input
+~~~~~
+.. onedal_class:: oneapi::dal::decision_forest::v1::infer_input
+
+
+.. _df_i_api_result:
+
+Result
+~~~~~~
+.. onedal_class:: oneapi::dal::decision_forest::v1::infer_result
+
+Operation
+~~~~~~~~~
+.. onedal_func:: oneapi::dal::v1::infer
```

Comment:
Need to describe this function "manually"

Reply:
added